server/pubEdgeProposal/api.js

```diff
@@ -0,0 +1,71 @@
+import fetch from 'node-fetch';
+
+import app, { wrap } from 'server/server';
+import { parseUrl } from 'utils/urls';
+import { isDoi } from 'utils/dois';
+
+import { getPubDataFromUrl } from './queries';
+
+const createPubEdgeProposalFromPub = (pub, url) => {
+	const { attributions, avatar, description, doi, lastPublishedAt, title } = pub;
+	const contributors = attributions.map((a) => a.name);
```

Comment:
(The comment below explains why this line is actually not needed, so the following is just FYI)

Not every `PubAttribution` object has a `name`. Those linked to a PubPub user will get their name info from the user model instead. Customarily we map attributions over `ensureUserForAttribution`, which will append a dummy user object to the attribution if one doesn't already exist, and then to look at `attribution.user.fullName`. Sounds a bit weird, I know — I am super open to a more robust tool for solving this problem!